src/java.base/share/classes/jdk/internal/reflect/ReflectionFactory.java

```diff
@@ -430,6 +437,86 @@ private final MethodHandle findReadWriteObjectForSerialization(Class<?> cl,
         }
     }
 
+    public final MethodHandle defaultReadObjectForSerialization(Class<?> cl) {
+        if (! isValidSerializable(cl)) {
+            return null;
+        }
+
+        return javaObjectStreamDefaultSupportAccess.defaultReadObject(cl);
+    }
+
+    public final MethodHandle defaultWriteObjectForSerialization(Class<?> cl) {
+        if (! isValidSerializable(cl)) {
+            return null;
+        }
+
+        return javaObjectStreamDefaultSupportAccess.defaultWriteObject(cl);
+    }
+
+    public final ObjectStreamField[] serialPersistentFields(Class<?> cl) {
+        if (! Serializable.class.isAssignableFrom(cl) || cl.isInterface() || cl.isEnum()) {
+            return null;
+        }
+
+        try {
+            Field field = cl.getDeclaredField("serialPersistentFields");
+            int mods = field.getModifiers();
+            if (! (Modifier.isStatic(mods) && Modifier.isPrivate(mods) && Modifier.isFinal(mods))) {
+                return null;
+            }
+            if (field.getType() != ObjectStreamField[].class) {
+                return null;
+            }
+            field.setAccessible(true);
```

Comment:
setAccessible() might need a doPriv to be successful in all cases.

Reply:
Other methods on this class do not use `doPrivileged()` so I thought it would be best to copy them; this, custom serialization libraries are using their own `doPrivileged()` blocks for those methods already. This could probably be safely changed (since a runtime permission is already required to access the user-facing `ReflectionFactory` class anyway), but would it be best to change it all at once? Or would it be better to leave it as is?